### DISCUSSION TITLE
"createSyncStoragePersister" stores everything under a single key.

### DISCUSSION THREAD

**Discussion by LevanKvirkvelia** [TanStack/query#5948] [Q&A] [answered]
### Description:
I recently started using createSyncStoragePersister for persistent storage and noticed an unexpected behavior. I had assumed that I could store data per query key, which would be particularly beneficial for managing large caches in React Native. However, it appears that everything is being stored under a single key.

### Expected Behavior:

Data should be stored separately for each unique query key.
When loading data, the function should fetch only the relevant data for the provided query key.

### Current Behavior:

All data, regardless of its original query key, is stored under one single key.
Consequently, when trying to load data, it retrieves everything stored under that single key, rather than the specific data associated with a given query key.

**Comment by TkDodo** [comment, marked as answer]
persisters save the whole QueryClient into a single location, that is on purpose.

we're working on a possibility to store each query separately here:
- #5125
